### PR TITLE
chore: create single endpoint to fetch all users transactions

### PR DESCRIPTION
**Pull request by Leona-cloud**

### CODE REVIEW

File: src/modules/api/transaction/services/index.ts [hunk 1]
```diff
@@ -788,6 +789,131 @@ export class TransactionService {
         });
     }
 
+    async getUserTransactions(options: UserTransactionHistoryDto) {
+        const paginationMeta: Partial<PaginationMeta> = {};
+
+        const startDate = startOfMonth(new Date(options.date));
```
Comment (calculusky) [inline src/modules/api/transaction/services/index.ts:795]: why are you filtering by date? Is that what is in the figma?

File: src/modules/api/transaction/controllers/v1/admin.ts [hunk 2]
```diff
@@ -142,4 +142,14 @@ export class AdminTransactionController {
             customerTransactionHistoryDto
         );
     }
+
+    @Get("user")
```
Comment (calculusky) [inline src/modules/api/transaction/controllers/v1/admin.ts:146]: pass userId as a path parameter

File: src/modules/api/transaction/services/index.ts (hunk 1 above)
Comment (Leona-cloud) [reply src/modules/api/transaction/services/index.ts:795]: According to your explanation yesterday, when the filer button is clicked there'll be an option to search via status/date . Also I was asked to follow the template for get all transactions

Comment (calculusky) [reply src/modules/api/transaction/services/index.ts]: Is there a date filtring on the transaction history API?

Comment (Leona-cloud) [reply src/modules/api/transaction/services/index.ts]: I've removed it

File: src/modules/api/transaction/services/index.ts [hunk 3]
```diff
@@ -789,15 +789,12 @@ export class TransactionService {
...
     }
 
-    async getUserTransactions(options: UserTransactionHistoryDto) {
+    async getUserTransactions(id: number, options: UserTransactionHistoryDto) {
         const paginationMeta: Partial<PaginationMeta> = {};
 
-        const startDate = startOfMonth(new Date(options.date));
-        const endDate = endOfMonth(new Date(options.date));
-
         const userExists = await this.prisma.user.findFirst({
```
Comment (calculusky) [inline src/modules/api/transaction/services/index.ts:795]: use findUnique

File: src/modules/api/transaction/services/index.ts [hunk 4]
```diff
@@ -829,18 +826,15 @@ export class TransactionService {
...
-                queryOptions.where.userId = +options.userId;
+                queryOptions.where.userId = id;
                 break;
             }
             case UserType.MERCHANT: {
-                queryOptions.where.OR = [
-                    { userId: +options.userId },
-                    { merchantId: +options.userId },
-                ];
+                queryOptions.where.OR = [{ userId: id }, { merchantId: id }];
```
Comment (calculusky) [inline src/modules/api/transaction/services/index.ts:837]: you can just fetch for merchants only. we will add the feature for sub-agents later

Comment (Leona-cloud) [reply src/modules/api/transaction/services/index.ts:833]: I've made an update to the PR

**Pull request merged**